std/container/dlist.d

```diff
@@ -702,8 +702,46 @@ Complexity: $(BIGOH r.walkLength)
     /// ditto
     alias stableLinearRemove = linearRemove;
 
+/**
+Removes the first occurence of an element from the list in linear time.
+
+Returns: True if the element existed and was successfully removed, false otherwise.
+
+Params:
+    value = value of the node to be removed
+
+Complexity: $(BIGOH n)
+     */
+    bool linearRemoveElement(T value)
+    {
+        auto n1 = findNodeByValue(_root, value);
+        // if node exists
+        if (n1)
+        {
+            auto n2 = n1._next._next;
+            BaseNode.connect(n1, n2);
+            return true;
+        }
+
+        return false;
+    }
+
+
 private:
 
+    BaseNode* findNodeByValue(BaseNode* n, T value)
+    {
+        assert(n, "Error: un-initialized List");
```

Comment:
shouldn't you just return null if the list is not initialized?

Reply:
I would also prefer to return null instead of asserting, though all the other `findNode*` functions do the same.